repos/spack_repo/builtin/packages/erf/package.py

```diff
@@ -0,0 +1,174 @@
+            self.define_from_variant("ERF_BUILD_TESTS", "tests"),
+            self.define_from_variant("ERF_BUILD_FCOMPARE", "fcompare"),
+            self.define_from_variant("ERF_ENABLE_FFT", "fft"),
+            self.define("ERF_DIM", "3"),
+        ]
+        args.append(self.define("ERF_USE_EXTERNAL_AMREX", False))
+        args.append(self.define("ERF_CLONE_AMREX", True))
+        args.append(self.define("GIT_SUBMODULE_PROTOCOL", "https"))
+        args.append(self.define("MPIEXEC_PREFLAGS", "--oversubscribe"))
+        args.append(self.define("CMAKE_BUILD_TYPE", "Release"))
+        args.append(self.define("ERF_DIM", "3"))
+
+        if "+netcdf" in self.spec:
+            args.extend(
+                [
+                    self.define("NetCDF_C_PATH", self.spec["netcdf-c"].prefix),
+                    self.define("NetCDF_FORTRAN_PATH", self.spec["netcdf-fortran"].prefix),
+                ]
+            )
+
+        if "+radiation" in self.spec:
+            if "+cuda" in self.spec:
+                args.append(self.define("Kokkos_ARCH_AMPERE80", True))
+            args.extend(
+                [
+                    self.define("ERF_ENABLE_KOKKOS", True),
+                    self.define("Kokkos_ARCH_AMPERE", True),
+                    self.define("ERF_ENABLE_EKAT", True),
+                    self.define("ERF_ENABLE_RRTMGP", True),
+                ]
+            )
+
+        if "+cuda" in self.spec:
+            archs = self.spec.variants["cuda_arch"].value or ["80"]
+            args.append(self.define("CMAKE_CUDA_ARCHITECTURES", ";".join(archs)))
+            args.extend(
+                [
+                    self.define("ERF_ENABLE_CUDA", "ON"),
+                    self.define("CUDAToolkit_ROOT", self.spec["cuda"].prefix),
+                ]
+            )
+
+        if "+hip" in self.spec:
+            args.append(self.define("CMAKE_HIP_ARCHITECTURES", "gfx906;gfx908;gfx90a"))
+
+        print(args)
+        return args
+
+    def setup_build_environment(self, env):
+        super(Erf, self).setup_build_environment(env)
+        env.set("GIT_CONFIG_COUNT", "1")
+        env.set("GIT_CONFIG_KEY_0", "url.https://github.com/.insteadOf")
+        env.set("GIT_CONFIG_VALUE_0", "git@github.com:")
+        env.set("AMREX_HOME", self.spec["amrex"].prefix)
+        if "+hip" in self.spec:
+            env.set("HIPCXX", self.spec["hip"].prefix.bin.hipcc)
+        if "+openmp" in self.spec:
+            if "%clang" in self.spec or "%gcc" in self.spec:
+                env.append_flags("CFLAGS", self.compiler.openmp_flag)
+                env.append_flags("CXXFLAGS", self.compiler.openmp_flag)
+                env.append_flags("FFLAGS", self.compiler.openmp_flag)
+            elif "%intel" in self.spec:
+                env.append_flags("CFLAGS", "-qopenmp")
+                env.append_flags("CXXFLAGS", "-qopenmp")
+                env.append_flags("FFLAGS", "-qopenmp")
+                print(self.compiler.openmp_flag)
```

Comment:
```suggestion
        if "+openmp" in self.spec:
            env.append_flags("CFLAGS", self["c"].openmp_flag)
            env.append_flags("CXXFLAGS", self["cxx"].openmp_flag)
            env.append_flags("FFLAGS", self["fortran"].openmp_flag)
```
If this can't be handled properly by the `intel-oneapi-compilers-classic` package (that provides the intel compiler) then we should modify that package to properly handle the `openmp_flag` option.

If that's necessary and you don't feel comfortable adding it to this PR yourself, let me know and I can work that in.